FBTweak/FBTweakCategory.m

```diff
@@ -34,40 +37,49 @@ - (instancetype)initWithName:(NSString *)name
 {
   if ((self = [super init])) {
     _name = [name copy];
-    
-    _orderedCollections = [[NSMutableArray alloc] initWithCapacity:4];
-    _namedCollections = [[NSMutableDictionary alloc] initWithCapacity:4];
+    _tweakCollections = [NSMutableArray array];
   }
   
   return self;
 }
 
 - (void)encodeWithCoder:(NSCoder *)coder
 {
-  [coder encodeObject:_name forKey:@"name"];
-  [coder encodeObject:_orderedCollections forKey:@"collections"];
+  [coder encodeObject:self.name forKey:@"name"];
+  [coder encodeObject:self.tweakCollections forKey:@"collections"];
 }
 
 - (FBTweakCollection *)tweakCollectionWithName:(NSString *)name
 {
-  return _namedCollections[name];
-}
-
-- (NSArray *)tweakCollections
-{
-  return [_orderedCollections copy];
+  NSUInteger index = [_tweakCollections indexOfObjectPassingTest:
+                      ^BOOL(FBTweakCollection * _Nonnull collection, NSUInteger __unused idx, BOOL * _Nonnull stop) {
+                        return [collection.name isEqualToString:name];
+  }];
+  return index != NSNotFound ? [self.tweakCollections objectAtIndex:index] : nil;
 }
 
 - (void)addTweakCollection:(FBTweakCollection *)tweakCollection
 {
-  [_orderedCollections addObject:tweakCollection];
-  [_namedCollections setObject:tweakCollection forKey:tweakCollection.name];
+  NSMutableArray<FBTweakCollection *> *collections = self.tweakCollections;
+  [collections addObject:tweakCollection];
+  self.tweakCollections = collections;
```

Comment:
* Why it's required? AFAIU `collections` and `self.teakCollections` are pointer to the same instance of `NSMutableArray`...
* Can following implementation of `addTweakCollection:` fit here?
  ```objc
  [self.tweakCollections addObject:tweakCollection];
  ```

Reply:
The problem here is that we now want the categories to be reactive. So the `tweakCollections` property should be KVO-compliant. Your suggestion does call the setter for `tweakCollections` and KVO does not fire.
There are two ways to make this work here:
1. Keep this implementations (or very similar) like
```
  [self.tweakCollections addObject:tweakCollection];
  self.tweakCollections = self.tweakCollections;
```

2. Not use mutable array and then just do
```
self.tweakCollections = [self.tweakCollections arrayByAddingObject:tweakCollection];
```

Reply:
Choose any of the two, and add a comment about it.

Reply:
Same below.